src/ndlbulkdownload/cli.py

```diff
@@ -43,6 +43,21 @@
 
 failed_urls = []
 
+# Global variable to store the output directory
+OUTPUT_DIR = None
+
+
+def check_and_set_output_dir(args):
+    global OUTPUT_DIR
+    if not os.path.exists(args.output_dir):
+        # Create the directory if it doesn't exist
+        os.makedirs(args.output_dir)
+        logging.info(f"Created directory: {args.output_dir}")
```

Comment:
Would prefer to see logging.debug() or perhaps not bother defining this at all since it's fairly trivial.

Rather than two log lines, there should only be one defining the outcome: user defined or the default location.

I'd prefer there's no additional branching.